migration/1.3.x-1.4.x/README.md

```diff
@@ -0,0 +1,94 @@
+but large files can only be processed by a single process. If data needs
+to be processed in parallel it must be spit into multiple smaller files,
+to help with this syntax has been added to the API to help split files.
+
+## Data
+
+1.4 adds the ability to use another PFS instance as a source for
+a `put-file` request. This allows you to migrate data from a running 1.3
+cluster into a new 1.4 cluster without bringing it down. The first step is
+getting both clusters running, if you're running them in the same
+Kubernetes cluster namespaces can be useful for avoiding name conflicts.
+Once both clusters are up and running you'll need to find an address for
+your 1.3 Pachyderm service so you can give it to the 1.4 cluster. This
+should be readily via `kubectl get all --namespace=<1.3 namespace>` Then
+to put data from a 1.3 repo into a 1.4 repo do:
+
+```
+pachctl put-file <1.4 repo> <1.4 branch> -f <1.3 ip address>:30650/<1.3 repo>/<1.3 branch> -r
+```
+
+Notice the `-r` parameter which tells Pachyderm to recursively scan the
+contents of the branch rather than just putting a single file, you can
+also migrate one file at a time by specifying it after the branch
+parameter in the address.
+
+Note that now is the time that you need to think about how you want to
+parallelize over your data. If your repo contains large files which were
+processed in 1.3 with `BLOCK` or `MAP` pipelines then you should use the
+`--split` flag in your `put-file` invocation to split the data up now.
+Refer to the [`put-file` docs](PLACEHOLDER) to learn more about `--split` works.
+
+After data has been migrated you can safely turn down the 1.3 cluster.
+
+## Pipelines
+
+### Inputs
+
+By far the biggest change to the Pipeline Spec is to the `inputs` field. Inputs
+no longer have a `method` field, instead they specify how their inputs can be
+parallelized using a glob pattern. [Read more about glob patterns
+here](PLACEHOLDER).
+
+The Partitions available in 1.3 match to glob patterns like so:
+
+| Partition | Glob   |
+| --------- | ------ |
+| `BLOCK`   | `/*/*` |
+| `FILE`    | `/*`   |
+| `REPO`    | `/`    |
+
+Note that to get the equivalent of `BLOCK` partition you should have put
+your files with `--split` in the step above. If your files are in a more
+deeply nested directory structure you'll need more layers of `*`s in your
+glob pattern. I.e. if you put the file `dir/foo` using `--split` rather
+than the file `foo` then you'll want to use `/*/*/*`.
+
+In 1.3 it wasn't possible to do `FILE` partitioning below the root layer
+of directories, in 1.4 this is possible with glob patterns. I.e. in the
+above example where a file is put to `dir/foo` with `--split` you could
+process all the pieces of the file together by passing `/*/*`.
+
+1.4 currently does not have a notion of incrementality, much of the
+incremental functionality present in 1.3 currently happens automatically.
+We'll be introducing a more advanced form of incrementality in a later
```

Comment:
Should we talk about how the "more advanced form of incrementality" is basically the equivalent of `/prev`?